network/p2p/unicast_manager.go

```diff
@@ -0,0 +1,25 @@
+package p2p
+
+import (
+	"context"
+
+	libp2pnet "github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/multiformats/go-multiaddr"
+
+	"github.com/onflow/flow-go/network/p2p/unicast/protocols"
+)
+
+// UnicastManager manages libp2p stream negotiation and creation, which is utilized for unicast dispatches.
+type UnicastManager interface {
+	// WithDefaultHandler sets the default stream handler for this unicast manager. The default handler is utilized
+	// as the core handler for other unicast protocols, e.g., compressions.
+	WithDefaultHandler(defaultHandler libp2pnet.StreamHandler)
+	// Register registers given protocol name as preferred unicast. Each invocation of register prioritizes the current protocol
+	// over previously registered ones.
+	Register(unicast protocols.ProtocolName) error
```

Comment:
If at all possible with reasonable effort, it would be great to document all error returns.

Reply:
https://github.com/onflow/flow-go/pull/3902/commits/5b643c9738cce3fed93ed64adfa690e924963c62